File: src/test/java/com/github/kpavlov/jreactive8583/example/TestUtil.java
package com.github.kpavlov.jreactive8583.example;

import java.util.function.BooleanSupplier;

import static org.assertj.core.api.Fail.fail;

public class TestUtil {

    private TestUtil() {
    }

    public static void waitFor(String message, BooleanSupplier condition) {
        int counter = 100;
        while (counter > 0 && !condition.getAsBoolean()) {
            try {
                Thread.sleep(100);
                counter--;
            } catch (InterruptedException e) {
                //noop
            }
        }
        if (counter == 0) {
            fail("Timeout: " + message);
        }
    }
}

File: src/test/java/com/github/kpavlov/jreactive8583/example/TestUtil.kt
package com.github.kpavlov.jreactive8583.example

import org.assertj.core.api.Fail

object TestUtil {

    @JvmStatic
    fun waitFor(message: String, block: () -> Boolean) {
        var counter = 100
        while (counter > 0 && !block.invoke()) {
            try {
                Thread.sleep(100)
                counter--
            } catch (e: InterruptedException) { //noop
            }
        }
        if (counter == 0) {
            Fail.fail<Any>("Timeout: $message")
        }
    }
}
